### PR TITLE
Load components with filter for removal

### PR DESCRIPTION
**Pull request by xipasduarte**
Adds a filter and allows for programatically stoping a given class from being executed.

### CODE REVIEW

File: wp-content/themes/genesis-starter/functions.php [hunk 1]
```diff
@@ -37,11 +37,20 @@
...
-	$styles->ready();
-	$shortcodes->ready();
+	$components = array(
+		'scripts'    => new \Patricia\Scripts(),
+		'shortcodes' => new \Patricia\Shortcodes(),
+		'styles'     => new \Patricia\Styles(),
+	);
+
+	// Excluded components
+	$exclude = \apply_filters( 'gs_exclude_components', array() );
```
Comment (goblindegook) [inline wp-content/themes/genesis-starter/functions.php]: Considering the starter theme is meant to be altered, will we need this?

Comment (xipasduarte) [inline wp-content/themes/genesis-starter/functions.php]: The idea would be to possibly filter the one or more components in specific pages. With this you don't need any conditions either on the classes themselves or on the pages.

It also allows for the usage of plugins to exclude things.

File: wp-content/themes/genesis-starter/functions.php [hunk 2]
```diff
@@ -37,11 +37,20 @@
...
-	$styles->ready();
-	$shortcodes->ready();
+	$components = array(
+		'scripts'    => new \Genesis_Starter\Scripts(),
+		'shortcodes' => new \Genesis_Starter\Shortcodes(),
+		'styles'     => new \Genesis_Starter\Styles(),
+	);
+
+	// Excluded components
+	$exclude = \apply_filters( 'gs_exclude_components', array() );
```
Comment (goblindegook) [inline wp-content/themes/genesis-starter/functions.php]: Why not filter the `$components` array instead? That would let you add as well as remove components.

Also, avoid abbreviations, call the hook `genesis_starter_components`.

File: wp-content/themes/genesis-starter/functions.php [hunk 3]
```diff
@@ -37,11 +37,20 @@
...
+
+	foreach ( $components as $name => $instance ) {
+		if ( in_array( $name, $exclude, true ) ) {
```
Comment (goblindegook) [inline wp-content/themes/genesis-starter/functions.php]: Filtering the `$components` array would make this unnecessary.

File: wp-content/themes/genesis-starter/functions.php [hunk 4]
```diff
@@ -37,11 +37,20 @@
...
+			continue;
+		}
+
+		$instance->ready();
```
Comment (goblindegook) [inline wp-content/themes/genesis-starter/functions.php]: ...although you should check that `$instance` implements `ready()` using `method_exists()`.

(You could force component classes to implement a common interface, but that creates cross-dependencies between theme and plugins that can crash the site if the theme is deactivated.)

**Comment by goblindegook** [comment]
Thanks!

**Pull request merged**